src/gatsby-remark-download-images.js

```diff
@@ -0,0 +1,52 @@
+const visit = require('unist-util-visit')
+const visitWithParents = require(`unist-util-visit-parents`)
+const path = require("path");
+const url = require('url');
+const fs = require('fs');
+const fsExtra = require("fs-extra")
+const slugify = require('slugify')
+var mime = require('mime-types');
+
+module.exports = async ({ markdownNode, markdownAST, getNode }) => {
+    
+    const frontmatter = markdownNode.frontmatter;
+    if (
+        frontmatter &&
+        frontmatter.title)
+    {
+            const dir = getNode(markdownNode.parent).dir;
+            const imagePath = path.join(dir, 'img');
+            
+            let remoteImages = [];
+
+            visit(markdownAST, [`image`], (node) => {
+              // I search for all the remote images in the mdx
+              if (node.url.startsWith('http')) {
+                remoteImages.push(node);
+              }
+            })
+
+            return Promise.all(
+              remoteImages.map( ( node ) => {
+                return new Promise( async function(resolve, reject) {
+
+                  const title = node.title || node.alt || node.url.slice(node.url.length - 10);
+                  const slugTitle = slugify(title, {remove: /[*+~.()'"!?:@,]/g});
+
+                  //I search img folder with slugtitle to find the file extension
+                  const files = await fsExtra.readdir(imagePath);
```

Comment:
How come you have to search the directory if you already have the exact image/node you’re processing?

Also where does this code actually download the image? Does it then change the mdx or would this have to be run every time before deploy?

Reply:
I was searching because in the download-images I only download the images, so during the build it doesn't know the extension of the image.

The image is downloaded using the script, not during the build.
It doesn't change the mdx, the AST is changed at every deploy.. Changing the mdx directly using the script would surely speed up the build! 
In that case 'gatsby-remark-download-images' could be eliminated

Reply:
The script that downloads the images is in src/scripts/download-images.js